library/ssl_srv.c

```diff
@@ -3830,7 +3830,7 @@ static int ssl_parse_client_psk_identity( mbedtls_ssl_context *ssl, unsigned cha
     n = ( (*p)[0] << 8 ) | (*p)[1];
     *p += 2;
 
-    if( n < 1 || n > 65535 || n > (size_t) ( end - *p ) )
+    if( !n || n > (uint16_t) ( end - *p ) )
```

Comment:
The cast of `end - *p` to `uint16_t` isn't right, is it? What if e.g. `end == *p + 65538`?

Reply:
Type cast to `uint16_t` is a serious mistake; it should have been casted to `size_t`.
But let's try to avoid type casts by rearranging comparison (the other comparison was changed to have the same style).